test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6953,6 +6953,29 @@ BOOST_AUTO_TEST_CASE(warn_about_suicide)
 	CHECK_WARNING(text, "\"suicide\" has been deprecated in favour of \"selfdestruct\"");
 }
 
+BOOST_AUTO_TEST_CASE(address_overload_resolution)
+{
+	char const* text = R"(
+		contract C {
+			function balance() returns (uint) {
+				this.balance; // to avoid pureness warning
+				return 1;
+			}
+			function transfer(uint amount) {
+				address(this).transfer(amount); // to avoid pureness warning
+			}
+		}
+		contract D {
+			function f() {
+				var x = (new C()).balance();
+				x;
```

Comment:
This produces "statement does not have an effect" warning. If you want to force "no warnings" I think you should use `CHECK_SUCCESS_NO_WARNINGS` or something like that.

Reply:
Ah, I thought `CHECK_SUCCESS_NO_WARNINGS` stands for success and not even warnings. Maybe it should read `CHECK_SUCCESS_IGNORE_WARNINGS `.

Reply:
Sure, that's what it stands for, but I thought that because of the comments above (`to avoid pureness warning`) you were aiming for a test without warnings.

Reply:
So it ignores warnings or doesn't?

Actually I don't mind the warnings myself in this code.